flytekit/core/base_task.py

```diff
@@ -128,7 +129,9 @@ def to_taskmetadata_model(self) -> _task_model.TaskMetadata:
         return _task_model.TaskMetadata(
             discoverable=self.cache,
             runtime=_task_model.RuntimeMetadata(
-                _task_model.RuntimeMetadata.RuntimeType.FLYTE_SDK, __version__, "python"
+                _task_model.RuntimeMetadata.RuntimeType.FLYTE_SDK,
+                __version__,
+                "sync_plugin" if self.use_sync_plugin else "python",
```

Comment:
This seems to be a quite fundamental change. It is a bit strange there is `sync_plugin` but no `async_plugin`.

Reply:
I think the intention of this change is for using `sync_plugin` in flyteplugins, `async_plugin` will be used by default.
Maybe we can change the variable to `use_api_task` or other variables?
How do you think?
cc @pingsutw

Reply:
Solved it, now we can specify it by lots of str value!